mobile/mobile-core/android-process-service.ts

```diff
@@ -25,6 +31,21 @@ export class AndroidProcessService implements Mobile.IAndroidProcessService {
 
 	public mapAbstractToTcpPort(deviceIdentifier: string, appIdentifier: string, framework: string): IFuture<string> {
 		return (() => {
+			if (this._shouldAddProcessExitEventListeners) {
```

Comment:
can you extract the whole `if` to a new private method, for example `tryAttachToProcessExitSignals`